starterAIs/starter.php

```diff
@@ -0,0 +1,162 @@
+    public const RICHNESS_BONUS = [0, 0, 2, 4];
+}
+
+class Board
+{
+    public $numberOfCells = 0;
+    public $cells = [];             // array[int idxCell] of Cell
+    public const MAX_CELLS = 37;
+
+    public function readInitInput(): void
+    {
+        $this->cells = [];
+        // $numberOfCells: 37
+        fscanf(STDIN, "%d", $this->numberOfCells);
+        for ($i = 0; $i < $this->numberOfCells; $i++) {
+            // $index: 0 is the center cell, the next cells spiral outwards
+            // $richness: 0 if the cell is unusable, 1-3 for usable cells
+            // $neigh0: the index of the neighbouring cell for each direction
+            fscanf(STDIN, "%d %d %d %d %d %d %d %d", $index, $richness, $nb0, $nb1, $nb2, $nb3, $nb4, $nb5);
+            $cell = new Cell();
+            $cell->index = $index;
+            $cell->richness = $richness;
+            $cell->neighbours = [$nb0, $nb1, $nb2, $nb3, $nb4, $nb5];
+            $this->cells[$index] = $cell;
+        }
+    }
+}
+
+class Tree
+{
+    public $cellIndex = 0;
+    public $size = 0;
+    public $isMine = false;
+    public $isDormant = false;
+    public const MAX_SIZE = 3;
+    public const HARVEST_SUNS = [0, 1, 2, 3];
+}
+
+class Action
+{
+    public $verb = self::WAIT;
+    public $index = 0;
+    public $indexTo = 0;
+    public $msg = "";
+    public const WAIT = 0;
+    public const SEED = 1;
+    public const GROW = 2;
+    public const COMPLETE = 3;
+    public const GROW_BASE_COSTS = [1, 3, 7];
+    public const COMPLETE_COST = 4;
+
+    public function toString(): string
+    {
+        // GROW cellIdx | SEED sourceIdx targetIdx | COMPLETE cellIdx | WAIT <message>
+        if ($this->verb == self::WAIT) {
+            $ans = "WAIT";
+        } elseif ($this->verb == self::SEED) {
+            $ans = "SEED " . $this->index . " " . $this->indexTo;
+        } elseif ($this->verb == self::GROW) {
+            $ans = "GROW " . $this->index;
+        } elseif ($this->verb == self::COMPLETE) {
+            $ans = "COMPLETE " . $this->index;
+        } else {
+            throw new \Exception("Invalid action");
+        }
+        if ($this->msg != "") {
+            $ans .= " " . $this->msg;
+        }
+        return $ans;
+    }
+}
+
+class Game
+{
+    public $board = null;           // Board
+    public $day = 0;
+    public $nutrients = 0;
+    public $mySun = 0;
+    public $myScore = 0;
+    public $oppSun = 0;
+    public $oppScore = 0;
+    public $oppIsWating = false;
+    public $numberOfTrees = 0;
+    public $trees = [];             // array[int idxCell] of Tree
+    public $possibleMoves = [];     // array of string
+    public $action = null;          // Action
+    public const MAX_DAY = 24;
+    public const EXTRA_POINT_PER_SUNS = 3;
+
+    public function readTurnInput(): void
+    {
+        // $day: the game lasts 24 days: 0-23
+        fscanf(STDIN, "%d", $this->day);
+        $this->startTime = microtime(true);
```

Comment:
I don't think this is needed it.

Reply:
I agree, starter AI should contain these more optional tools

Reply:
yes, starting timer remained in the code by mistake, I deleted it